src/Development/IDE/Core/Shake.hs

```diff
@@ -164,6 +180,11 @@ data ShakeExtras = ShakeExtras
     ,exportsMap :: Var ExportsMap
     -- | A work queue for actions added via 'runInShakeSession'
     ,actionQueue :: ActionQueue
+    , hiedb :: HieDb -- ^ Use only to read.
+    , hiedbWriter :: HieDbWriter -- ^ use to write
```

Comment:
Read and write what exactly? What is the hiedb used for? Source files in external packages or also in local packages?

Reply:
Both. References are looked up in local packages, and definitions are looked up in both.